dotCMS/src/main/java/com/dotcms/rest/api/v1/contenttype/ContentTypeResource.java

```diff
@@ -181,34 +188,39 @@ public Response updateType(@PathParam("id") final String id, final ContentTypeFo
 	@NoCache
 	@Produces({MediaType.APPLICATION_JSON, "application/javascript"})
 	public Response deleteType(@PathParam("id") final String id, @Context final HttpServletRequest req)
-			throws DotDataException, DotSecurityException, JSONException {
+			throws DotDataException, JSONException {
 
 		final InitDataObject initData = this.webResource.init(null, true, req, true, null);
 		final User user = initData.getUser();
 
 		ContentTypeAPI tapi = APILocator.getContentTypeAPI(user, true);
 
-
-		ContentType type = null;
 		try {
-			type = tapi.find(id);
-		} catch (NotFoundInDbException nfdb) {
+
+			ContentType type = null;
 			try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)